utils/cryptography.go

```diff
@@ -3,51 +3,51 @@ package oyster_utils
 import (
 	"crypto/aes"
 	"crypto/cipher"
-	"crypto/rand"
-	"crypto/sha256"
 	"encoding/hex"
 	"hash"
-	"strings"
-
-	"golang.org/x/crypto/pbkdf2"
 )
 
-func deriveKey(passphrase string, salt []byte) ([]byte, []byte) {
-	if salt == nil {
-		salt = make([]byte, 8)
-		// http://www.ietf.org/rfc/rfc2898.txt
-		// Salt.
-		rand.Read(salt)
-	}
-	return pbkdf2.Key([]byte(passphrase), salt, 1000, 32, sha256.New), salt
-}
-
-func Encrypt(passphrase, plaintext string) string {
-	key, salt := deriveKey(passphrase, nil)
-	iv := make([]byte, 12)
-	// http://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-38d.pdf
-	// Section 8.2
-	rand.Read(iv)
-	b, _ := aes.NewCipher(key)
-	aesgcm, _ := cipher.NewGCM(b)
-	data := aesgcm.Seal(nil, iv, []byte(plaintext), nil)
-	return hex.EncodeToString(salt) + "-" + hex.EncodeToString(iv) + "-" + hex.EncodeToString(data)
+func Encrypt(key string, secret string, nonce string) []byte {
+	keyInBytes, err := hex.DecodeString(key)
+	panicOnErr(err)
+	block, err := aes.NewCipher(keyInBytes)
+	panicOnErr(err)
+	gcm, err := cipher.NewGCM(block)
+	panicOnErr(err)
+	nonceInBytes, err := hex.DecodeString(nonce[0 : 2*gcm.NonceSize()])
+	panicOnErr(err)
+	data := gcm.Seal(nil, nonceInBytes, []byte(secret), nil)
+	return data
```

Comment:
There is a lot of calls to this same panicOnErr function in this file.  Almost every step in the encrypting/decrypting process has the opportunity for errors and I didn't want to ignore them.  I am open to doing this differently if we want.

Reply:
This seems fine to me, at least until we've done some more testing with these changes to ensure it's stable.